chore(InlineEdit): animate inline edit to edit off

lee-chase opened the pull request:
Contributes to #1492 

Add a moment of delight, animate between edit and edit off icons.

#### What did you change?

Rebuilt SVG for Edit16 and used it to animate to an icon like EditOff16

#### How did you test and verify your work?
Storybook toggle `disabled` control

dcwarwick commented on packages/cloud-cognitive/src/components/InlineEdit/edit-n-edit-off-svg.js:
```diff
@@ -0,0 +1,76 @@
+      <g fill="currentColor">
+        {/* image */}
+        <g mask="url(#line-mask">
+          <path
```
```suggestion
          <path
            className="pencil"
```
See note above about including the pencil fill into the transition for a smoother effect.

dcwarwick reviewed (changes requested):
This is very nice. I've got a few organisational suggestions, and one to make the transition a little smoother (the pen colour doesn't currently transition, while the underline colour does). I notice that apart from animating the icon beautifully the `disabled` prop on InlineEdit doesn't actually work at the moment -- but I presume you know that already :-)

dcwarwick commented on packages/cloud-cognitive/src/components/InlineEdit/_edit-n-edit-off-svg.scss:
```diff
@@ -0,0 +1,32 @@
+  transition: all $duration--moderate-02 motion(standard, productive);
+}
+
+[disabled] > .edit-n-edit-off-svg .underline {
```
PS I do see that several of the related Carbon styles key off the `disabled` attribute in this same way. I don't think I like that either :-) unless there's some good reason for it that I haven't thought of.

lee-chase replied:
This was very much as a result of seeing button and tooltip use of `[disabled]`. Can't see a good reason in code to do so.

lee-chase commented:
Thanks, @dcwarwick, meant to leave this as draft for a design review.

Addressed your suggestions pending comment from @sdignum

When we spoke yesterday she seemed to think the animation was unlikely to be used and hence not needed.

I'll leave here, just in case anyone has a good use case for this animation.
